pennylane_qiskit/qiskit_device2.py

```diff
@@ -556,6 +565,11 @@ def _execute_runtime_service(self, circuits, session):
 
         results = []
 
+        ### ToDo: ensure the measurements in circuit.measurements commute
+        ### Need to wait to implement transforms such as split non-commute, ham expand, etc.
+        ### [SC-62047]
+        ### Can't raise a warning/error because if we could figure out that the measurements
+        ### did not commute then we could just split it instead of raising a warning/error.
```

Comment:
I put this comment here instead of in the pre-processing step to emphasize that bugs that occur in `_execute_runtime_service` (specifically there) are a direct result of the measurements not commuting/other issues. 

For example if a user decides to set `use_primitives=False` and have a circuit that measures `qml.expval(qml.Z(0)), qml.expval(qml.X(0)), qml.expval(qml.Y(0))`, this will run without any error or warning but produce an inaccurate answer.

The point of the comment is to say that I would like to raise an error or warning in the event that the measurement is going to produce inaccurate results i.e. not diagonalized properly. But to note that if I could check if the diagonalization was incorrect then I could also diagnose what about it was incorrect and split the measurement instead of raising a warning/error. 

Right now I'm thinking about raising a warning that is a bit more general. For example, if there are measurement processes acting on the same wire, could raise a warning that says something like "Make sure these commute or else they might be inaccurate!!"

This warning falls out of the PR's scope, however, so I did not implement it but left it as a comment.

Reply:
This feature branch won't be merged into `master` until that support is added, and users shouldn't be using feature branches that are being actively developed, so there's no need for any user-facing accommodations like warnings for that. 

>But to note that if I could check if the diagonalization was incorrect then I could also diagnose what about it was incorrect and split the measurement instead of raising a warning/error.

You shouldn't split the measurements here, because the post-processing knows what shape to expect based on what comes out of `preprocessing`. Anything that would split things into different tapes needs to occur in the `preprocessing` function so that the shapes of the output tuples from `execute` are as expected.

This function shouldn't worry about things like that at all - it should assume that any inputs it gets are valid, because pre-processing and validation is completed before this part of the execution pipeline.

Reply:
Ah I edited my comment cause I realized that was a bit weird to say but yeah, I wasn't intending to split the measurements there. I think I can clarify the comment a bit to mean something more like "Right now this function assumes that the inputs are valid, but since there are some transforms missing this isn't the case." Sort of a note to self type of comment.